src/wp-includes/feed-rss2-comments.php

```diff
@@ -67,44 +75,52 @@
-				$title = get_the_title($comment_post->ID);
-				/** This filter is documented in wp-includes/feed.php */
-				$title = apply_filters( 'the_title_rss', $title );
-				/* translators: Individual comment title. 1: Post title, 2: Comment author name */
-				printf(ent2ncr(__('Comment on %1$s by %2$s')), $title, get_comment_author_rss());
-			} else {
-				/* translators: Comment author title. 1: Comment author name */
-				printf(ent2ncr(__('By: %s')), get_comment_author_rss());
-			}
-		?></title>
-		<link><?php comment_link() ?></link>
-		<dc:creator><![CDATA[<?php echo get_comment_author_rss() ?>]]></dc:creator>
-		<pubDate><?php echo mysql2date('D, d M Y H:i:s +0000', get_comment_time('Y-m-d H:i:s', true, false), false); ?></pubDate>
-		<guid isPermaLink="false"><?php comment_guid() ?></guid>
-<?php if ( post_password_required($comment_post) ) : ?>
-		<description><?php echo ent2ncr(__('Protected Comments: Please enter your password to view comments.')); ?></description>
-		<content:encoded><![CDATA[<?php echo get_the_password_form() ?>]]></content:encoded>
-<?php else : // post pass ?>
-		<description><![CDATA[<?php comment_text_rss() ?>]]></description>
-		<content:encoded><![CDATA[<?php comment_text() ?>]]></content:encoded>
-<?php endif; // post pass
-	/**
-	 * Fires at the end of each RSS2 comment feed item.
-	 *
-	 * @since WP-2.1.0
-	 *
-	 * @param int $comment->comment_ID The ID of the comment being displayed.
-	 * @param int $comment_post->ID    The ID of the post the comment is connected to.
-	 */
-	do_action( 'commentrss2_item', $comment->comment_ID, $comment_post->ID );
-?>
+		<title>
+				<?php
+				if ( ! is_singular() ) {
+					$title = get_the_title( $comment_post->ID );
+					/** This filter is documented in wp-includes/feed.php */
+					$title = apply_filters( 'the_title_rss', $title );
+					/* translators: Individual comment title. 1: Post title, 2: Comment author name */
+					printf( ent2ncr( __( 'Comment on %1$s by %2$s' ) ), $title, get_comment_author_rss() );
+				} else {
+					/* translators: Comment author title. 1: Comment author name */
+					printf( ent2ncr( __( 'By: %s' ) ), get_comment_author_rss() );
+				}
+				?>
+		</title>
+		<link><?php comment_link(); ?></link>
+		<dc:creator><![CDATA[<?php echo get_comment_author_rss(); ?>]]></dc:creator>
+		<pubDate><?php echo mysql2date( 'D, d M Y H:i:s +0000', get_comment_time( 'Y-m-d H:i:s', true, false ), false ); ?></pubDate>
+		<guid isPermaLink="false"><?php comment_guid(); ?></guid>
+			<?php if ( post_password_required( $comment_post ) ) : ?>
+		<description><?php echo ent2ncr( __( 'Protected Comments: Please enter your password to view comments.' ) ); ?></description>
+		<content:encoded><![CDATA[<?php echo get_the_password_form(); ?>]]></content:encoded>
+			<?php else : // post pass ?>
+		<description><![CDATA[<?php comment_text_rss(); ?>]]></description>
+		<content:encoded><![CDATA[<?php comment_text(); ?>]]></content:encoded>
+				<?php
```

Comment:
Why is this one so indented?

Reply:
Some layout errors from the nest PHP code, updated PR with manual fix to follow.

Reply:
PR updated